Undo backend ignores the working directory argument

```
$ go run scripts/make.go test -v -s proc -r TestWorkDir -b undo
=== RUN   TestWorkDir
--- FAIL: TestWorkDir (1.73s)
    support.go:246: enabling recording for github.com/undoio/delve/pkg/proc_test.TestWorkDir
    proc_test.go:81: recording
    proc_test.go:83: replaying
    proc_test.go:2547: Expected /tmp got /home/grees/go/src/github.com/undoio/delve/pkg/proc
```
The problem is that `UndoRecord` in pkg/proc/gdbserial/undo.go ignores the `wd` (working directory) argument. Compare with `LLDBLaunch` in pkg/proc/gdbserial/gdbserver.go, which has:
```
if wd != "" {
        proc.Dir = wd
}
```

--- a/pkg/proc/gdbserial/undo.go
+++ b/pkg/proc/gdbserial/undo.go
@@ -54,6 +54,10 @@ func UndoRecord(cmd []string, wd string, quiet bool) (recording string, err erro
 	args = append(args, cmd...)
 	lrcmd := exec.Command("live-record", args...)
 
+	if wd != "" {
+		lrcmd.Dir = wd
+	}
+
 	if !quiet {
 		lrcmd.Env = os.Environ()
 		// Setting UNDO_debug_level alone is not enough

